### ISSUE TITLE
Feature request: support some GraphViz features

### ISSUE THREAD

**Issue by DavidBJaffe** [nadavrot/layout#13]
Hi -- this crate is really cool!!

I tested an example

```
digraph G {
node [width=0.13,height=0.13,fontsize=10,shape=point];
edge [fontsize=18,penwidth=1.3,arrowsize=1.3,fontname=Arial];
rankdir=LR;
labeljust=l;
margin=0;
15 -> 10 [minlen=2,color=red,label="0 (len=173,supp=807)"];
5 -> 15 [minlen=2,color=red,label="1 (len=430,supp=1567)"];
10 -> 7 [minlen=1,color=black,label="2 (len=26,supp=541)"];
7 -> 10 [minlen=1,color=black,label="3 (len=30,supp=951)"];
0 -> 7 [minlen=2,color=red,label="4 (len=618,supp=3220)"];
14 -> 13 [minlen=2,color=red,label="5 (len=632,supp=269)"];
11 -> 15 [minlen=1,color=black,label="6 (len=43,supp=40)"];
}
```

and got this output

<img width="1382" alt="image" src="https://user-images.githubusercontent.com/12055080/206863542-1651034c-3fa3-4f64-b26a-5bd5d33a4dec.png">

as compared to what running dot yields

<img width="1306" alt="image" src="https://user-images.githubusercontent.com/12055080/206863671-e2e1c6e3-b337-4cd0-8794-7aa1e70493de.png">

It would be awesome to get closer to the dot output.

Thank you very much!!

**Comment by nadavrot** [comment]
There are a few things that layout can do better. First, it does not support he shape 'point'. Second, the edge crosses one of the labels. Third, I agree that it would be better to round some of the edges (the support for that is limited).

**Comment by nadavrot** [comment]
@DavidBJaffe were there charts that the program rendered well?

**Comment by DavidBJaffe** [comment]
Well I have a lot of graphs that look kind of like the above one (and which are generated by another program).  I think it's pretty good as is.

**Comment by dolezvo1** [comment]
Hi, I made a separate issue regarding subgraph support (#37), but it appears the rendering does not handle basic properties either.

```dot
digraph G {
  subgraph cluster_0 {
    style=filled;
    color=lightgrey;
    node [style=filled,color=white];
    a0 -> a1 -> a2 -> a3;
    label = "process #1";
  }

  subgraph cluster_1 {
    node [style=filled];
    b0 -> b1 -> b2 -> b3;
    label = "process #2";
    color=blue
  }
  start -> a0;
  start -> b0;
  a1 -> b3;
  b2 -> a3;
  a3 -> a0;
  a3 -> end;
  b3 -> end;

  start [shape=Mdiamond];
  end [shape=Msquare];
  a1 [shape=box, width=10, height=4];
  b1 [shape=box, width=10, height=5];
}
```

Graphviz gives me this:

![Image](https://github.com/user-attachments/assets/fc8a53b0-4f33-4817-9649-b8290b2d0e86)


Meanwhile `layout` gives me this (I manually enhanced the background on the right side):

![Image](https://github.com/user-attachments/assets/0c9a193d-b0d9-43ec-9946-401d023e4f55) ![Image](https://github.com/user-attachments/assets/ca4b62ae-44af-4873-b9ff-a1eb99dec423)

To summarize, it appears to me that using `width` and `height` with layout doesn't get interpreted correctly. If I'm not mistaken, I think it gets added to the stroke thickness (which makes the edges appear to be wrong), but nothing else.

**Comment by mert-kurttutan** [comment]
@dolezvo1 I am working on subgraph feature. Since the DAG data structure does not truly represent subgraph hierarchy, the overwriting rules are wrong when it comes to node features. I will provide update once the branch is ready so you can test your cases.

**Comment by dolezvo1** [comment]
That's awesome, thank you!
